File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TDBXInt64Value.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Wrap the Int64 type and allows it to be null
 *
 */

public class TDBXInt64Value extends DBXValue {

	protected boolean ValueNull = false;
	private long DBXInt64Value;
	
	public TDBXInt64Value() {
		super();
		setDBXType(DBXDataTypes.Int64Type);
	}
	
	public boolean isNull() {
		return ValueNull;
	}

	public void setNull() throws DBXException {
		ValueNull = true;
		DBXInt64Value = 0;
	}
	
	
	public void SetAsInt64(long Value) throws DBXException {
		DBXInt64Value = Value;
		ValueNull = false;
	}

	public long GetAsInt64() throws DBXException {
		return DBXInt64Value;
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TDBXInt64Value.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXInt64Value.h"


@implementation TDBXInt64Value
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:Int64Type];
	}
	return self;
}
-(void) dealloc{
	[super dealloc];
}
-(void) SetNull {
	ValueNull = YES;
	DBXInternalValue = 0;
}
-(bool) isNull {
	return ValueNull;
}
-(void) SetAsInt64:(long long)value{
	DBXInternalValue = value;
}
-(long long) GetAsInt64{
	return DBXInternalValue;
}


@end
